encoders/numeric/CompressionVaeEncoder/__init__.py

```diff
@@ -16,50 +19,60 @@ class CompressionVaeEncoder(BaseNumericEncoder, BaseTFEncoder):
-                 initializer='lecun_normal', batch_size=32, batch_size_test=32,
-                 feature_normalization=False, tb_logging=False,
+    def __init__(self, model_path: str = None,
                  *args, **kwargs):
         """
-        :param model_path: specifies the path to the model, and also acts as the model name.
-        :param X : array, shape (n_samples, n_features)
-            Training data for the VAE.
-            Alternatively, X can be the path to a root-directory containing npy files (potentially nested), each
-            representing a single feature vector. This allows for handling of datasets that are too large to fit
-            in memory.
-            Can be None (default) only if a model with this name has previously been trained. Otherwise None will
-            raise an exception.
-        :param train_valid_split: controls the random splitting into train and test data. Here 99% of X is used
-        for training, and only 1% is reserved for validation.
-        :param dim_latent: specifies the dimensionality of the latent space.
-        :param iaf_flow_length: controls how many IAF(Inverse Autoregressive Flow) flow layers the model has.
-        :param cells_encoder: determines the number, as well as size of the encoders fully connected layers.
-        :param initializer: controls how the model weights are initialized, with options being `orthogonal` (default),
-        `truncated_normal`, and `lecun_normal`
-        :param batch_size: determine the batch sizes used for training.
-        :param batch_size_test: determine the batch sizes used for testing.
-        :param feature_normalization: tells CVAE whether it should internally apply feature normalization
-        (zero mean, unit variance, based on the training data) or not. If True, the normalisation factors are stored
-        with the model and get applied to any future data.
-        :param tb_logging: determines whether the model writes summaries for TensorBoard during the training process.
+        :param model_path: specifies the path to the pretrained model
+
         """
         super().__init__(*args, **kwargs)
-        self.X = X
-        self.train_valid_split = train_valid_split
-        self.output_dim = output_dim
-        self.iaf_flow_length = iaf_flow_length
-        self.cells_encoder = cells_encoder
-        self.initializer = initializer
-        self.batch_size = batch_size
-        self.batch_size_test = batch_size_test
         self.model_path = model_path
-        self.feature_normalization = feature_normalization
-        self.tb_logging = tb_logging
 
     def post_init(self):
         super().post_init()
-        self.to_device()
+        from cvae import cvae
+        import cvae.lib.model_iaf as model
+        import tensorflow as tf
+
+        if self.model_path and os.path.exists(self.model_path):
+            params_path = os.path.join(self.model_path, 'params.json')
```

Comment:
You can skip the check on `model_path` and do it for `params_path`:

```python
params_path = os.path.join(self.model_path, 'params.json') if self.model_path else None

if params_path and os.path.exists(self.model_path):
...
else:
raise PretrainedModelFileDoesNotExist()
```
```

Reply:
I think it would be better assigning to params_path an existent model path or None, and check if the JSON file is really contained in the folder afterwards:

```.py
params_path = os.path.join(self.model_path, 'params.json') if self.model_path and os.path.exists(
            self.model_path) else None

if self.params_path and os.path.exists(params_path):
...
else:
raise PretrainedModelFileDoesNotExist()
```

because otherwise, we don't know if the folder contains that file. And if the JSON for any reason in missing we are going to get an Exception when we are loading it.

Reply:
perfect! It seems good!

Reply:
Nice! I'm going to push that change.